python/bindings/include/openravepy/pybind11/pybind11_bindings.h

```diff
@@ -241,45 +241,77 @@ class OPENRAVEPY_API PyOpenRAVEException : public py::object {
 
     /// Uses cpp_function's return_value_policy by default
     template <typename ... Extra>
-    PyOpenRAVEException &def_property_static(const char *name, const py::cpp_function &fget, const py::cpp_function &fset, const Extra& ... extra) {
+    PyOpenRAVEException &def_property_static(const char *name,
+                                             const py::cpp_function &fget,
+                                             const py::cpp_function &fset,
+                                             const Extra &...extra) {
+        static_assert(0 == py::detail::constexpr_sum(std::is_base_of<py::arg, Extra>::value...),
+                      "Argument annotations are not allowed for properties");
         auto rec_fget = get_function_record(fget), rec_fset = get_function_record(fset);
-        char *doc_prev = rec_fget->doc; /* 'extra' field may include a property-specific documentation string */
-        py::detail::process_attributes<Extra ...>::init(extra ..., rec_fget);
-        if (rec_fget->doc && rec_fget->doc != doc_prev) {
-            free(doc_prev);
-            rec_fget->doc = strdup(rec_fget->doc);
+        auto *rec_active = rec_fget;
```

Comment:
`auto *`  made me think an intent behind it. can we simply use `py::detail::function_record *` for these local variables for simplicity?

Reply:
@yoshikikanemoto I'm really sorry that I did not mentioned I copied this code from https://github.com/pybind/pybind11/blob/v2.13.6/include/pybind11/pybind11.h#L1819

Reply:
However I can take the least change: https://github.com/rdiankov/openrave/pull/1526